benchmark/httpapi.py

```diff
@@ -100,6 +112,104 @@ def delete(self, id):
             return fail(ResultNotFound())
 
 
+@implementer(IBackend)
+class TxMongoBackend(object):
+    """
+    The backend that uses txmongo driver to work with MongoDB.
+    """
+    def __init__(self, hostname="127.0.0.1", port=27017):
+        connection = MongoConnectionPool(host=hostname, port=port)
+        self.collection = connection.benchmark.results
```

Comment:
It creates a connection object, a database object and a collection object.
But no actual connection is established at that point (i.e. no network packets are sent), that is done behind the scenes when the collection object gets used.